File: maths/cycle1/fiches-methodes3e/I/C/m0104.c
#include <stdio.h>
#include <stdlib.h>
#include "display.h"

int
main() {
    int p = 0;
    puts("# Exercice 4 :");
    puts("# -----------");
    puts("# On donne un entier n. Donner deux multiples consécutifs de n.");
    puts("-----------------------------------------------------------------------------------");
    puts("");
    p = display_wording();
    if (p == 0) {
        wording();
    }
    p = put_up_correct_version();
    if (p == 0) {
        display_result();
    }
    exit(EXIT_SUCCESS);
}

File: maths/cycle1/fiches-methodes3e/I/Pascal/m0104.pas
program m0104;

{$mode objfpc}{$H+}

uses
    method01, display;

var
    q1 : Integer = 4;
    q2 : Integer;
    bool : Boolean;

begin
    WriteLn('# Exercice 4 :');
    WriteLn('# -----------');
    WriteLn('# On donne un entier n. Donner deux multiples consécutifs de n.');
    WriteLn('-----------------------------------------------------------------------------------');
    WriteLn();

    bool := DisplayMethodWording;
    if bool then PutAMethod();

    bool := PutUpCorrectVersion;
    if not bool then exit;

    q2 := q1 + 1;
    WriteLn(q1, ' x n et ', q2, ' x n sont deux multiples consécutifs de l''entier n.');
    WriteLn('Plus généralement, étant donné deux entiers non nuls q1 et q2');
    WriteLn('tels que q2 = q1 + 1, q1 x n et q2 x n sont deux multiples consécutifs de n.');

    {$IFDEF WINDOWS}
    ReadLn;
    {$ENDIF}
end.

